filters/apiusagemonitoring/spec.go

```diff
@@ -290,58 +286,61 @@ func (s *apiUsageMonitoringSpec) buildClientTrackingInfo(apiIndex int, api *apiC
 	}
 }
 
-// generateRegExpStringForPathTemplate normalizes the given path template and
-// creates a regular expression from it.
-func generateRegExpStringForPathTemplate(pathTemplate string) (normalizedPathTemplate, matcher, pathLabel string) {
-	pathParts := strings.Split(pathTemplate, "/")
-	matcherPathParts := make([]string, 0, len(pathParts))
-	normalizedPathTemplateParts := make([]string, 0, len(pathParts))
-	pathLabelParts := make([]string, 0, len(pathParts))
-	for _, p := range pathParts {
-		if p == "" {
-			continue
-		}
-		placeholderName := getVarPathPartName(p)
-		if placeholderName == "" {
-			// this part is not a placeholder: match it exactly
-			matcherPathParts = append(matcherPathParts, p)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, p)
-			pathLabelParts = append(pathLabelParts, p)
-
-		} else {
-			// this part is a placeholder: match a wildcard for it
-			matcherPathParts = append(matcherPathParts, regexUrlPathPart)
-			normalizedPathTemplateParts = append(normalizedPathTemplateParts, ":"+placeholderName)
-			pathLabelParts = append(pathLabelParts, "{"+placeholderName+"}")
-		}
-	}
-	rawRegEx := &strings.Builder{}
-	rawRegEx.WriteString("^")
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString(strings.Join(matcherPathParts, `\/`))
-	rawRegEx.WriteString(regexOptionalSlashes)
-	rawRegEx.WriteString("$")
+var (
+	regexpMultipleSlashes   = regexp.MustCompile(`/+`)
+	regexpLeadingSlashes    = regexp.MustCompile(`^/*`)
+	regexpTrailingSlashes   = regexp.MustCompile(`/*$`)
+	regexpMiddleSlashes     = regexp.MustCompile(`([^/^])/+([^/*])`)
```

Comment:
if I am not mistaken, we can just replace every occurence of multiple consecutive slashes. Then this expression could become simply: `/{2,}`

Reply:
Yes `{2,}` is exactly the meaning of `+`, however, I would always prefer the latter.

Reply:
I didn't mean `/+` -> `/{2,}` (the meaning of `+` is `{1,}`. What i meant was: `([^/^])/+([^/*])` -> `/{2,}`. Then the following expression could be changed like this. From:

```
regexpMiddleSlashes.ReplaceAllString(path, "$1/+$2")
```

to:

```
regexpMiddleSlashes.ReplaceAllString(path, "")
```

Which is somewhat simpler.

Reply:
As far as I can see your suggestion would remove all occurrences of more than one two slash, meaning there is no slash left. However, I want to replace any number of slashes in the middle against a regular expression pattern `\+`.

I agree, that my condition looks a bit difficult, because I need to identify middle slashes (which requires the regular expression to be context sensitive) after I already did some replacements for leading and trailing slashes. May the expression and the whole algorithm can be simplified by first replacing all slashes against `\+` and than modifying the first and the last to match `\*`.

I think, I give this a try.

Reply:
The changes would look as follows: 

* leading slashes `^/*` to `^/*\+?`
* trailing slashes `/*$` to `(|/+\+?)$`
* middle slashes `([^/^])/+([^/*])` to `/+`

The reason for the increasing complexity in leading and trailing slashes is, that they are reused for creating the template. The change would also lead to unexpected side effects on the template, that I could compensate by reintroducing the before regular expressions in addition.

@aryszka what do you think.

Reply:
I see. Thanks for the explanation.